files/commit-hieradata.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/sh
+
+readonly cluster_id=$1
+readonly branch="origin/tf/lbaas/${cluster_id}"
+
+cd appuio_hieradata || exit 1
+
+git config user.name "${GIT_AUTHOR_NAME}"
+git config user.email "${GIT_AUTHOR_EMAIL}"
+
+amend=0
+# Checkout feature branch
+# 1. try to check out as tracking branch from upstream
+# 2. checkout as new branch
+# 3. checkout existing local branch
+# For existing local branch, amend existing commit
+if ! git checkout -t "${branch}"; then
+  if ! git checkout -b "${branch}"; then
+    git checkout "${branch}"
+    amend=1
+  fi
+fi
+
+git add "lbaas/${cluster_id}.yaml"
+
+status=$(git status --porcelain)
+echo "'${status}'"
+
+commit_message="Update LBaaS hieradata for ${cluster_id}"
+push=1
+if [ "${status}"  = "M  lbaas/${cluster_id}.yaml" ]; then
+  if [ "$amend" -eq 1 ]; then
+    git commit --amend --no-edit
+  else
+    git commit -m"${commit_message}"
+  fi
+elif [ "${status}" != "" ]; then
+  # assume new hieradata
+  commit_message="Create LBaaS hieradata for ${cluster_id}"
+  git commit -m "${commit_message}"
+else
+  push=0
+fi
+
+if [ "${push}" -eq 1 ]; then
+  push_args=
+  if [ "${amend}" -eq 1 ]; then
+    push_args="--force"
+  fi
+  git push ${push_args} origin "${branch}"
+  # Set local branch to track upstream branch at origin
+  git branch -u "origin/${branch}"
```

Comment:
Is there a reason not to put the `-u` option on `git push`?

Reply:
I just forgot `push -u` is a thing, fixed in the latest version